### PR TITLE
Refactor assets loading

### PR DESCRIPTION
**Pull request by XhmikosR**
* move fonts first
* move CSS and JS along with the rest
* move default auto theme media checks to the HEAD instead of imports which are blocking

All CSS/JS files are better grouped together. This allows the browser to plan ahead better.

Also, add `defer` for scripts. This makes the DOM load faster (~250ms down to ~25ms).

---

## Thank you for your contribution to the Pi-hole Community!

Please read the comments below to help us consider your Pull Request.

We are all volunteers and completing the process outlined will help us review your commits quicker.

**Please make sure you**

 1. Base your code and PRs against the repositories developmental branch.
 2. [Sign Off](https://docs.pi-hole.net/guides/github/how-to-signoff/) all commits as we enforce the [DCO](https://docs.pi-hole.net/guides/github/dco/) for all contributions
 3. [Sign](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) all your commits as they must have verified signatures
 4. File a pull request for any change that requires changes to [our documentation](https://docs.pi-hole.net/) at our [documentation repo](https://github.com/pi-hole/docs) 

---

**What does this PR aim to accomplish?:**

**How does this PR accomplish the above?:**

**Link documentation PRs if any are needed to support this PR:**

---
**By submitting this pull request, I confirm the following:**

1. I have read and understood the [contributors guide](https://docs.pi-hole.net/guides/github/contributing/), as well as this entire template. I understand which branch to base my commits and Pull Requests against.
2. I have commented my proposed changes within the code and I have tested my changes.
3. I am willing to help maintain this change if there are issues with it later.
4. It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
5. I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))
6. I have checked that another pull request for this purpose does not exist.
7. I have considered, and confirmed that this submission will be valuable to others.
8. I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
9. I give this submission freely, and claim no ownership to its content.

---
- [x] I have read the above and my PR is ready for review. *Check this box to confirm*

### CODE REVIEW

**Comment by XhmikosR** [comment]
Let's keep this on hold for now, it seems we are hitting some edge case so I want to test the branch more

**Review by yubiuser — COMMENTED** [review]
Interestingly, the DOM Loading Time only improves on Firefox for me, not Chrome.

Before


After Chrome


After Firefox

**Comment by XhmikosR** [comment]
Yeah, I'm not using Chrome but it shouldn't be slower. I guess Chrome already does better/other optimizations and thus the gains might not be the same.

**Comment by XhmikosR** [comment]
Also, assuming @DL6ER can help with #3302, this should also help in theory. And I have some other ideas for later like preloading some stuff.

**Comment by XhmikosR** [comment]
I converted this to draft. I'll rebase after #3300 and it would be nice to see the difference on Chrome then.

**Review by yubiuser — COMMENTED** [review]
FF development


FF this branch
ff_pr

Chrome development
chrome_dev

Chrome this branch
chrome_pr

**Comment by XhmikosR** [comment]
I don't think it should be slower on Chrome and the difference is statistically insignificant. Did you run it many times?

Either way, I still think it's a good change: now the JS files are non-blocking. We can look at doing further improvements later with preloading if needed.

**Comment by DL6ER** [comment]
Tested and seems to be working fine.

One question though: Why are we using `defer` and not `async` here? Wouldn't the latter be potentially faster if we start executing scripts as soon as possible even before then entire page is parsed? I don't see any issues here as - without either `defer` or `async` (so what we have right now), the scripts are downloaded and executed immediately, blocking any further parsing until the script is completed.

**Comment by XhmikosR** [comment]
Because with `async` you have no guarantee about the script order, so everything would be probably broken :)

File: scripts/lua/header.lp [hunk 1]
```diff
@@ -102,20 +99,26 @@ if startsWith(scriptname, 'groups') then
...
+    <link rel="stylesheet" href="<?=pihole.fileversion('vendor/waitMe/waitMe.min.css')?>">
     <link rel="stylesheet" href="<?=pihole.fileversion('vendor/select2/select2.min.css')?>">
     <link rel="stylesheet" href="<?=pihole.fileversion('vendor/adminLTE/AdminLTE.min.css')?>">
 
-    <!-- Theme styles -->
+    <!-- Theme styles (<?= theme.name ?>) -->
     <link rel="stylesheet" href="<?=pihole.fileversion('style/pi-hole.css')?>">
-    <link rel="stylesheet" href="<?=pihole.fileversion('style/themes/'..theme.name..'.css')?>">
+<? if theme.name == "default-auto" then ?>
+    <link rel="stylesheet" href="<?=pihole.fileversion('style/themes/default-dark.css')?>" media="(prefers-color-scheme: dark)">
```
Comment (XhmikosR) [inline scripts/lua/header.lp:109]: This saves 2 blocking CSS requests: first the default-auto.css had to be downloaded and then the appropriate CSS theme style.

**Review by DL6ER — APPROVED** [review]
